source/h2_connection.c

```diff
@@ -1865,6 +1907,12 @@ static void s_move_stream_to_thread(
     }
 
     if (has_outgoing_data) {
+        if (aws_linked_list_empty(&connection->thread_data.outgoing_streams_list)) {
+            /* transition from nothing to write -> something to write */
+            uint64_t now_ns = 0;
+            aws_channel_current_clock_time(connection->base.channel_slot->channel, &now_ns);
+            connection->thread_data.outgoing_timestamp_ns = now_ns;
+        }
         aws_linked_list_push_back(&connection->thread_data.outgoing_streams_list, &stream->node);
```

Comment:
consider moving this to a helper function so that https://github.com/awslabs/aws-c-http/pull/338 can take advantage of it

Reply:
will move when we feel like need to